Dag37.RabbitFramework/RabbitFramework/Publishers/CommandPublisher.cs

```diff
@@ -34,19 +34,26 @@ public CommandPublisher(IBusProvider busProvider)
 
             var waitHandle = new ManualResetEvent(false);
             string responseJson = null;
+            bool isError = false;
 
-            _commandCallbacks[correlationId] = (response) =>
+            _commandCallbacks[correlationId] = (response, responseIsError) =>
             {
                 responseJson = response;
+                isError = responseIsError;
                 waitHandle.Set();
             };
 
             PublishCommandMessage(message, queueName, key, correlationId);
 
             bool gotResponse = await Task.Run(() => waitHandle.WaitOne(timeout));
 
+            if (isError)
+            {
+                throw new CommandPublisherException(responseJson);
```

Comment:
Misschien handig om hier de CommandPublisherException zelf de serializen/deserializen. Je zou dan in de CommandPublisher.InnerException de daadwerkelijke command kunnen stoppen aan de listener kant

Reply:
Wellicht zou je de daadwerkelijke exception al direct kunnen serializen zonder tussenkomst van een CommandPublisherException

Reply:
Dan krijg je meer overhead, terwijl je ook gewoon de responsejson kan pakken als het een error blijkt te zijn

Reply:
Je hoeft alleen de message te returnen, zo doet HttpClient dat dacht ik ook
alleen dan op een iets andere manier

Reply:
Ehm ik bedoelde exception, niet command